### PR TITLE
docs: Updated FHIR Works readme

### PR DESCRIPTION
**Pull request by shyogesh-sw**
Issue #, if available:

Description of changes: Updated the FHIR works readme. Please review and provide your feedback/approve.

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,197 +1,212 @@
...
+The AuthN/Z providers are defined in `package.json` and `config.ts`. You can choose  appropriate providers. SMART allows greater granularity into authentication than Cognito and is the FHIR standard. It allows you to access a FHIR record only if that record has reference to the user. FHIR Works on AWS controls the working of your FHIR server.
 
-## Capabilities
 
-This deployment implementation utilizes Lambda, DynamoDB, S3 and Elasticsearch to provide these FHIR capabilities:
+## FHIR on AWS features
```
Comment (rsmayda) [inline README.md]:
```suggestion
## FHIR Works on AWS features
```

File: README.md [hunk 2]
```diff
@@ -1,197 +1,212 @@
...
+The AuthN/Z providers are defined in `package.json` and `config.ts`. You can choose  appropriate providers. SMART allows greater granularity into authentication than Cognito and is the FHIR standard. It allows you to access a FHIR record only if that record has reference to the user. FHIR Works on AWS controls the working of your FHIR server.
```
Comment (nguyen102) [inline README.md]: Is this line needed?
`FHIR Works on AWS controls the working of your FHIR server.`

Maybe I'm missing something, but I'm not sure what that additional line is adding.

File: README.md [hunk 3]
```diff
@@ -1,197 +1,212 @@
...
-## Capabilities
 
-This deployment implementation utilizes Lambda, DynamoDB, S3 and Elasticsearch to provide these FHIR capabilities:
+## FHIR on AWS features
 
-- CRUD operations for all R4 or STU3 base FHIR resources
-- Search capabilities per resource type
-- Ability to do versioned reads (vread)
-- Ability to post a transaction bundle of 25 entries or less
+FHIR Works on AWS solution utilizes AWS Lambda, Amazon DynamoDB, Amazon S3 and Amazon Elasticsearch Service to provide the following FHIR features:
```
Comment (nguyen102) [inline README.md]:
```suggestion
FHIR Works on AWS utilizes AWS Lambda, Amazon DynamoDB, Amazon S3 and Amazon Elasticsearch Service to provide the following FHIR features:
```

File: README.md [hunk 4]
```diff
@@ -1,197 +1,212 @@
...
 
-## Quick start/installation
++ Create, Read, Update, Delete (CRUD) operations for all R4 or STU3 base FHIR resources
++ Search capabilities per resource type
++ Ability to do versioned reads ([vread](https://www.hl7.org/fhir/http.html#vread))
++	Ability to post a transaction bundle of 25 entries or less. Presently, transaction bundles with only 25 entries or less are supported.
```
Comment (nguyen102) [inline README.md]: Delete the extra space at the beginning of this line?

File: README.md [hunk 5]
```diff
@@ -1,197 +1,212 @@
...
+1. Clone or download the repository to a local directory.
+ 
+Example:
 
 ```sh
 git clone https://github.com/awslabs/fhir-works-on-aws-deployment.git
 ```
 
-### Install
+**Note**: To modify FHIR Works on AWS, create your own fork of the GitHub repository. This allows you to check in any changes you make to your private copy of the solution.
```
Comment (nguyen102) [inline README.md]: Nitpick: Change `solution` to `code`.

My thought being, we want to use the term `solution` only when referring to the actual [FHIR Solution](https://aws.amazon.com/solutions/implementations/fhir-works-on-aws/)

File: README.md [hunk 6]
```diff
@@ -1,197 +1,212 @@
...
 
-[Postman](https://www.postman.com/) is an API Client for RESTful services that can run on your development desktop for making requests to the FHIR Server. Postman is highly suggested and will make accessing the FHRI API much easier.
+**Importing the collection file**
 
-Included in this code package, under the folder “postman”, are JSON definitions for some requests that you can make against the server. To import these requests into your Postman application, you can follow the directions [here](https://kb.datamotion.com/?ht_kb=postman-instructions-for-exporting-and-importing). Be sure to import the collection file.
+Under the Postman folder, you can access the JSON definitions for some API requests that you can make against the server. To import these requests into your Postman application, click [here](https://kb.datamotion.com/?ht_kb=postman-instructions-for-exporting-and-importing). 
 
-> [Fhir.postman_collection.json](./postman/Fhir.postman_collection.json)
+**Note**: Ensure that you import the `[Fhir.postman_collection.json](./postman/Fhir.postman_collection.json)` collection file.
```
Comment (nguyen102) [inline README.md]:
```suggestion
**Note**: Ensure that you import the [Fhir.postman_collection.json](./postman/Fhir.postman_collection.json) collection file.
```

File: README.md [hunk 7]
```diff
@@ -1,197 +1,212 @@
...
+**Setting up environment variables**
 
-- Fhir_Local_Env.json
-- Fhir_Dev_Env.json
-- Fhir_Prod_Env.json
+Set up the following three environment files:
 
-The variables required in the POSTMAN collection can be found in `Info_Output.yml` or by running `serverless info --verbose`
++ `Fhir_Local_Env.json`
++	`Fhir_Dev_Env.json`
```
Comment (nguyen102) [inline README.md]: Remove the extra space at the beginning of line 96 and 97?

File: README.md [hunk 8]
```diff
@@ -1,197 +1,212 @@
...
-- API_KEY: from Service Information: api keys: developer-key
-- CLIENT_ID: from Stack Outputs: UserPoolAppClientId
-- AUTH_URL: `https://<CLIENT_ID>.auth.<REGION>.amazoncognito.com/oauth2/authorize`
+For instructions on importing the environment JSON, click [here](https://thinkster.io/tutorials/testing-backend-apis-with-postman/managing-environments-in-postman).
 
-To know what all this FHIR API supports please use the `GET Metadata` postman to generate a [Capability Statement](https://www.hl7.org/fhir/capabilitystatement.html).
+The following variables required in the Postman collection can be found in `Info_Output.yml` or by running `serverless info --verbose`:
 
-### Authorizing a user
++	`API_URL: from Service Information:endpoints: ANY`
```
Comment (nguyen102) [inline README.md]: Remove extra space?

File: README.md [hunk 9]
```diff
@@ -1,197 +1,212 @@
...
++	`API_KEY: from Service Information: api keys: developer-key`
++ `CLIENT_ID: from Stack Outputs: UserPoolAppClientId`
++ `AUTH_URL: https://<CLIENT_ID>.auth.<REGION>.amazoncognito.com/oauth2/authorize`
```
Comment (nguyen102) [inline README.md]:
```suggestion
+ API_URL: from Service Information:endpoints: ANY
+ API_KEY: from Service Information: api keys: developer-key
+ CLIENT_ID: from Stack Outputs: UserPoolAppClientId
+ AUTH_URL: https://<CLIENT_ID>.auth.<REGION>.amazoncognito.com/oauth2/authorize
```

File: README.md [hunk 10]
```diff
@@ -1,197 +1,212 @@
...
 
-FHIR Works on AWS solution uses role based access control (RBAC) to determine what operations and what resource types the requesting user has access too. The default ruleset can be found here: [RBACRules.ts](src/RBACRules.ts). For users to access the API they must use an OAuth access token. This access token must include scopes of either:
+To find what the FHIR Server supports, use the `GET Metadata` postman to generate a [Capability Statement](https://www.hl7.org/fhir/capabilitystatement.html).
```
Comment (nguyen102) [inline README.md]:
```suggestion
To find what the FHIR Server supports, use the `GET Metadata` Postman request to retrieve the [Capability Statement](https://www.hl7.org/fhir/capabilitystatement.html).
```

File: README.md [hunk 11]
```diff
@@ -1,197 +1,212 @@
...
 
-- `openid profile` Must have both
-- `aws.cognito.signin.user.admin`
+**Authorizing a user**
 
-Using either of the above scopes will include the user groups in the access token.
+FHIR Works on AWS solution uses Role-Based Access Control (RBAC) to determine what operations and what resource types the requesting user has access too. The default ruleset can be found in [RBACRules.ts](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/mainline/src/RBACRules.ts). To access the API, you must use the OAuth access token. This access token must include scopes of either `openid`, `profile` or `aws.cognito.signin.user.admin`.
```
Comment (nguyen102) [inline README.md]:
```suggestion
FHIR Works on AWS uses Role-Based Access Control (RBAC) to determine what operations and what resource types the requesting user has access to. The default rule set can be found in [RBACRules.ts](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/mainline/src/RBACRules.ts). To access the API, you must use the OAuth access token. This access token must include scopes of either `openid`, `profile` or `aws.cognito.signin.user.admin`. 
```

File: README.md [hunk 12]
```diff
@@ -1,197 +1,212 @@
...
-3. Click `Get New Access Token`
-4. A sign in page should pop up where you should put in your username and password (if you don't know it look at the [init-auth.py](scripts\init-auth.py) script)
-5. Once signed in the access token will be set and you will have access for ~1 hour
+For more information, click [here](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-app-idp-settings.html).
 
-#### Retrieving an access token via script (scope = aws.cognito.signin.user.admin)
+**Retrieving access token via postman using the openid profile**
 
-A Cognito OAuth access token can be obtained using the following command substituting all variables with their values from `INFO_OUTPUT.yml` or the previously mentioned `serverless info --verbose` command.
+To access the FHIR API, an access token is required. This can be obtained following these steps within Postman:
```
Comment (nguyen102) [inline README.md]:
```suggestion
To access the FHIR API, an access token is required. This can be obtained by following these steps within Postman:
```

File: README.md [hunk 13]
```diff
@@ -1,197 +1,212 @@
...
 
-For Windows:
+1. Open Postman and choose the operation (for example, `GET Patient`).
+2. In the **Authorization** tab, choose **Get New Access Token**.
+4. A sign in page appears. Enter the username and password (if you don't know it look at the [init-auth.py](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/mainline/scripts%5Cinit-auth.py) script).
+5. After signing in, the access token is set and you have the access for approximately one hour.
```
Comment (nguyen102) [inline README.md]:
```suggestion
3. A sign in page appears. Enter the username and password (if you don't know it look at the [init-auth.py](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/mainline/scripts%5Cinit-auth.py) script).
4. After signing in, the access token is set and you have the access for approximately one hour.
```

File: README.md [hunk 14]
```diff
@@ -1,197 +1,212 @@
...
-  ![Postman Body Request Settings](resources/postman_body_request_settings.png)
++ When making a `POST`/`PUT` request to the server, if you get an error that includes the text `Failed to parse request body as JSON resource`, check that you've set the request headers correctly. The header for `Content-Type` should be either `application/json` or `application/fhir+json` If you're using Postman for making requests, in the Body tab, be sure to also set the setting to `raw` and `JSON`.
+![Postman Body Request Settings](resources/postman_body_request_settings.png)
 
 ## Feedback
-
-We'd love to hear from you! Please reach out to our team: [fhir-works-on-aws-dev](mailto:fhir-works-on-aws-dev@amazon.com) for any feedback.
+We'd love to hear from you! Please reach out to our team: [fhir-works-on-aws-dev](mailto:fhir-works-on-aws-dev@amazon.com) for any feedback.
```
Comment (nguyen102) [inline README.md]: I believe in our retro we decided to remove our email. Let's rephrase this to have them file GH issues if they have any questions or concerns.

File: README.md (hunk 2 above)
Comment (shyogesh-sw) [reply README.md]: Deleted.

File: README.md (hunk 1 above)
Comment (shyogesh-sw) [reply README.md]: thanks. Fixed

File: README.md (hunk 3 above)
Comment (shyogesh-sw) [reply README.md]: Fixed

File: README.md (hunk 4 above)
Comment (shyogesh-sw) [reply README.md]: Fixed.

File: README.md (hunk 5 above)
Comment (shyogesh-sw) [reply README.md]: fixed.

File: README.md (hunk 6 above)
Comment (shyogesh-sw) [reply README.md]: fixed. Removed the quotes

File: README.md (hunk 7 above)
Comment (shyogesh-sw) [reply README.md]: corrected

File: README.md (hunk 9 above)
Comment (shyogesh-sw) [reply README.md]: corrected

File: README.md (hunk 10 above)
Comment (shyogesh-sw) [reply README.md]: fixed

File: README.md (hunk 11 above)
Comment (shyogesh-sw) [reply README.md]: Changed this line from " FHIR Works on AWS uses Role-Based Access Control (RBAC) to determine what operations and what resource types the requesting user has access to.

to "FHIR Works on AWS uses Role-Based Access Control (RBAC) to determine what operations and what resource types a user can access."

File: README.md (hunk 12 above)
Comment (shyogesh-sw) [reply README.md]: fixed.

File: README.md (hunk 13 above)
Comment (shyogesh-sw) [reply README.md]: fixed

File: README.md (hunk 14 above)
Comment (shyogesh-sw) [reply README.md]: fixed

**Review by nguyen102 — APPROVED** [review]

**Comment by shyogesh-sw** [comment]
Readme restructured and updated
